woocommerce/google-listings-and-ads
# Fix attributes in variants for attribute mapping

### Changes proposed in this Pull Request:

This PR Fixes reported bugs getting taxonomy and custom attributes based sources in Attribute Mapping on Variations. 


### Detailed test instructions:


**Testing Taxonomy based Attributes** 

1. Create a product variant with a Taxonomy based variations, like Color.
2. Create an attribute mapping rule as (target Color, source Product Color,  Apply to ALL)
3. Save
4. Sync your product (using connection test is ok)
5. See the Color attribute is being loaded in Merchant.
6. Create a simple product.
7. Add an attribute Color (or several colors) to it
8. Sync that product
9. See the Color attribute (first one) is being loaded in Merchant.

**Testing Custom based Attributes** 

Use this snippet to create a custom attribute: 

```php
add_filter(
		'woocommerce_gla_attribute_mapping_sources_custom_attributes',
		function( $values ) {
			return array_merge( $values, ['material']);
		}
	);
```

1. Create a product variant with a Custom attribute named "material" based variations.
2. Create an attribute mapping rule as (target Material, source material,  Apply to ALL)
3. Save
4. Sync your product (using connection test is ok)
5. See the Material attribute is being loaded in Merchant.
6. Create a simple product.
7. Add an attribute material
8. Sync that product
9. See the Material attribute is being loaded in Merchant.


### Additional details:


### Changelog entry

> Fix - Bug in Attribute Mapping  - with Taxonomy based rules not being applied in variations.
